lib/ansible/modules/storage/netapp/na_cdot_snapshot.py

```diff
@@ -0,0 +1,238 @@
+            name : Name of the smapshot
+            volume_name: Name of the containing volume
+
+        :return: Details about the snapshot. None if not found.
+        :rtype: dict
+        """
+        snapshot_list = netapp_utils.zapi.NaElement('snapshot-get-iter')
+        snapshot_info = netapp_utils.zapi.NaElement('snapshot-info')
+        snapshot_info.add_new_child('name', self.name)
+        snapshot_info.add_new_child('volume', self.volume_name)
+
+        query = netapp_utils.zapi.NaElement('query')
+        query.add_child_elem(snapshot_info)
+
+        snapshot_list.add_child_elem(query)
+
+        result = self.server.invoke_successfully(snapshot_list, True)
+
+        return_value = None
+
+        if result.get_child_by_name('num-records') and \
+                int(result.get_child_content('num-records')) >= 1:
+
+            snapshot_info = result.get_child_by_name(
+                'attributes-list').get_child_by_name(
+                    'snapshot-info')
+            # check if snapshot name & volume match
+            returned_name = snapshot_info.get_child_content('name')
+            returned_volume = snapshot_info.get_child_content('volume')
+
+            if returned_name == self.name and returned_volume == self.volume_name:
+                return_value = {
+                    'name': self.name,
+                    'volume_name': self.volume_name,
+                }
+
+        return return_value
+
+    def create_snapshot(self):
+        snapshot_create = netapp_utils.zapi.NaElement.create_node_with_children(
+            'snapshot-create', **{'volume': self.volume_name,
+                                  'snapshot': self.name})
+
+        try:
+            self.server.invoke_successfully(snapshot_create,
+                                            enable_tunneling=True)
+        except netapp_utils.zapi.NaApiError as e:
+            self.module.fail_json(msg='Error creating snapshot %s on %s: %s' % (self.name, self.volume_name, to_native(e)),
+                                  exception=traceback.format_exc())
+
+    def delete_snapshot(self):
+        snapshot_delete = netapp_utils.zapi.NaElement.create_node_with_children(
+            'snapshot-delete', **{'volume': self.volume_name,
+                                  'snapshot': self.name})
+
+        try:
+            self.server.invoke_successfully(snapshot_delete,
+                                            enable_tunneling=True)
+        except netapp_utils.zapi.NaApiError as e:
+            self.module.fail_json(msg='Error deleting snapshot %s on %s: %s' % (self.name, self.volume_name, to_native(e)),
+                                  exception=traceback.format_exc())
+
+    def apply(self):
+        changed = False
+        snapshot_exists = False
+        snapshot_detail = self.get_snapshot()
+
+        if snapshot_detail:
+            snapshot_exists = True
+
+            if self.state == 'absent':
+                changed = True
+
+        else:
```

Comment:
215-216 can be changed into `elif self.state == 'present':`

Reply:
Right, was also a leftover from the copied module.